### PR TITLE
Remove ObjectMeta.ClusterName usage

### LINKED ISSUE #2173: CAPZ should not use ObjectMeta.ClusterName
/kind bug

There is an upstream discussion about dropping ObjectMeta.ClusterName: https://github.com/kubernetes/kubernetes/pull/108717. Related CAPI issue: https://github.com/kubernetes-sigs/cluster-api/issues/6305

This field is not populated by Kubernetes and we should not use it as an CAPZ Cluster Name.

This issue is about doing some research where we use it in CAPZ and fix it if it is used.

h/t https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/3321

/kind bug

### PR DESCRIPTION
**Pull request by DannyBrito**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Remove usage of ObjectMeta.ClusterName in CAPZ

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2173

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: controllers/helpers_test.go [hunk 1]
```diff
@@ -259,7 +259,7 @@ func TestReconcileAzureSecret(t *testing.T) {
 
 	cluster.Default()
 	azureCluster.Default()
-	azureCluster.ClusterName = "testCluster"
+	azureCluster.Name = "testCluster"
```
Comment (CecileRobertMichon) [inline controllers/helpers_test.go:262]: I believe this should be `cluster.Name`

File: controllers/helpers_test.go [hunk 2]
```diff
@@ -293,7 +293,7 @@ func TestReconcileAzureSecret(t *testing.T) {
 			g.Expect(err).NotTo(HaveOccurred())
 			g.Expect(cloudConfig.Data).NotTo(BeNil())
 
-			if err := reconcileAzureSecret(context.Background(), kubeclient, owner, cloudConfig, azureCluster.ClusterName); err != nil {
+			if err := reconcileAzureSecret(context.Background(), kubeclient, owner, cloudConfig, azureCluster.Name); err != nil {
```
Comment (CecileRobertMichon) [inline controllers/helpers_test.go:296]:
```suggestion
			if err := reconcileAzureSecret(context.Background(), kubeclient, owner, cloudConfig, cluster.Name); err != nil {
```

File: controllers/helpers_test.go (hunk 1 above)
Comment (CecileRobertMichon) [reply controllers/helpers_test.go:262]: basing this of the fact that we're using `clusterScope.ClusterName()` outside tests which maps to `cluster.Name`

**Comment by mboersma** [comment]
I looked to see [how CAPA fixed this](https://github.com/kubernetes/kubernetes/pull/108717/files). Wow, I'm glad CAPZ doesn't maintain an external swagger spec and wasn't really using `ObjectMeta.ClusterName`.

**Comment by mboersma** [comment]
Also, I did a separate audit of the CAPZ codebase (using regex `\.ClusterName[^\(]` and then searching for explicit `ObjectMeta` struct initializations), and this was the only usage I found that needed to change.

**Review by mboersma — APPROVED** [review]
/lgtm

**Review by mboersma — APPROVED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

thank you @DannyBrito!

**Pull request merged**